src/components/ui/email-input/email-input.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { UseFormReturn } from 'react-hook-form';
+import { InputValidationError } from '../input-validation-error';
+
+type Props = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  form: UseFormReturn<any, any, any>;
+};
+
+export const EmailInput = ({ form }: Props) => {
+  const {
+    register,
+    formState: { errors },
+  } = form;
+
+  return (
+    <div>
+      <label htmlFor="email" className="block text-sm/6 font-medium text-gray-900">
+        Email address
+      </label>
+      <div className="mt-2">
+        <input
+          id="email"
+          {...register('email', {
+            required: true,
+            pattern:
+              /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
```

Comment:
move this regExp to constants folder in `regExp.ts`

Reply:
fixed